sapi/fpm/fpm/fpm_php.c

```diff
@@ -87,8 +87,10 @@ int fpm_php_apply_defines_ex(struct key_value_s *kv, int mode) /* {{{ */
 
 	if (!strcmp(name, "extension") && *value) {
 		zval zv;
+		zend_interned_strings_switch_storage(0);
 		php_dl(value, MODULE_PERSISTENT, &zv, 1);
-		return Z_TYPE(zv) == IS_TRUE;
+		zend_interned_strings_switch_storage(1);
+		return Z_TYPE(zv) == IS_TRUE ? 2 : 0;
```

Comment:
Tricky ``-1``, ``0`` and ``2`` need some explanation.
I understood the meaning from the diff, but it's gong to be more difficult if this's merged.

Reply:
Good point. I added macros for return values so it should be more clear now.